charon/src/transform/mod.rs

```diff
@@ -110,6 +111,8 @@ pub static INITIAL_CLEANUP_PASSES: &[Pass] = &[
     NonBody(&normalize::expand_associated_types::Transform),
     // `--remove-adt-clauses`: Remove all trait clauses from type declarations.
     NonBody(&simplify_output::remove_adt_clauses::Transform),
+    // Transform Drops into Calls to drop_in_place.
+    UnstructuredBody(&normalize::desugar_drops::Transform),
```

Comment:
Please move this pass before `skip_trait_refs_when_known` so that we can benefit from its normalization of trait method calls.